deps/rabbit/scripts/rabbitmq-server

```diff
@@ -147,5 +148,6 @@ else
     # force that statement to succeed and the signal handler to properly
     # execute. Because the statement below has an exit code of 0, the
     # signal handler has to restate the expected exit code.
-    wait "$rabbitmq_server_pid" || true
+    wait "$rabbitmq_server_pid" && exit_code=$? || exit_code=$?
+    exit "$exit_code"
```

Comment:
```suggestion
    set +e
    wait "$rabbitmq_server_pid"
    exit "$?"
```

Reply:
`set -e` / `set +e` are modifying global state, so we'll have to remember that it's only safe if we change it just before exiting (may still affect traps though) or set it back to `-e` right after wait which bit only if it was actually toggled before that.

Reply:
>  we'll have to remember that it's only safe if we change it just before exiting

Yes, in the case of the `stop_rabbitmq_server` function, that's what happens.

Reply:
I agree. If we are to unset `-e`, we should restore it after the wait and save the exit code and return it explicitly.

I should have noted more details about that dash(1) behavior, because I can't remember enough today... Thus I don't remember how to test any regressions :-/